### ISSUE TITLE
Debian Wheezy owncloud-client 1.8.0 fails certificate check

### ISSUE THREAD

**Issue by boltronics** [owncloud/client#2979]
I just upgraded the client from 1.7.1 to 1.8.0 via the official ownCloud repository:
```
deb http://download.opensuse.org/repositories/isv:/ownCloud:/desktop/Debian_7.0/ /
```

Now when I open ownCloud, I get the pop-up:

```
Warnings about current SSL Connection:
The issuer certificate of a locally looked up certificate could not be found
The root CA certificate is not trusted for this purpose
with Certificate StartCom Class 2 Primary Intermediate Server CA
Organization: StartCom Ltd.
Unit: Secure Digital Certificate Signing
Country: IL
Fingerprint (MD5): 35:60:c6:d7:6a:78:ce:1e:e8:12:6c:5f:80:b9:d5:04
Fingerprint (SHA1): a1:ac:e4:04:6b:6e:33:22:32:b8:7e:cf:b6:f3:7a:07:63:72:01:47

Effective Date: Wed Oct 24 20:57:09 2007
Expiration Date: Tue Oct 24 20:57:09 2017
Issuer: StartCom Certification Authority
Organization: StartCom Ltd.
Unit: Secure Digital Certificate Signing
Country: IL
```

This is the relevant part of the client log:
```
03-18 14:30:06:006 0x210f630 !!! OCC::CheckServerJob created for  QUrl( "https://owncloud.systemsaviour.com/" )  querying "status.php" 
03-18 14:30:07:877 0x210f630 FolderMan: Syncing is disabled, no scheduling. 
03-18 14:30:08:063 0x210f630 SslHandshakeFailedError:  "SSL handshake failed"  : can be caused by a webserver wanting SSL client certificates 
03-18 14:30:08:064 0x210f630 void OCC::AbstractNetworkJob::slotFinished() 6 "SSL handshake failed" 
03-18 14:30:08:064 0x210f630 AccountState connection status change:  "Undefined" -> "Status not found" 
03-18 14:30:08:064 0x210f630 AccountState state change:  "Disconnected" -> "NetworkError" 
03-18 14:30:08:064 0x210f630 Disabling sync scheduler, terminating sync 
03-18 14:30:08:065 0x210f630 error: status.php replied  0 "" 
03-18 14:30:09:906 0x210f630 Client is on latest version! 
03-18 14:30:11:685 0x210f630 Saving  0  unknown certs. 
03-18 14:30:11:686 0x210f630 void OCC::SyncJournalDb::close() "/home/abolte/ownCloud/.csync_journal.db" 
03-18 14:30:11:686 0x210f630 No database Transaction to commit 
03-18 14:30:11:686 0x210f630 SocketApi:  dtor 
owncloud.log (END)
```

Here are the results of the certificate installation test:
https://www.sslshopper.com/ssl-checker.html?host=owncloud.systemsaviour.com#hostname=owncloud.systemsaviour.com

I can confirm the fingerprints do match my CA-provided intermediate certificate (startcom-sub.class2.server.ca.pem). I run Apache, and have the following certificates defined:

```
SSLCertificateFile /certpath/systemsaviour.com.crt
SSLCertificateKeyFile /certpath/systemsaviour.com.key
SSLCertificateChainFile /certpath/startcom-sub.class2.server.ca.pem
SSLCACertificateFile /certpath/startcom-ca.pem
```

```apache2ctl -t``` says ```Syntax OK```, which is expected since the problem only started happening after receiving the owncloud-client update. Evidentially, the ownCloud client is not detecting the certificate referenced by the SSLCACertificateFile directive - but I would think it should be using the locally installed certificates anyway? I don't have any certificate issues browsing the owncloud installation in either Chromium or Firefox.

**Comment by spitzlei** [comment]
Same Problem here. SSLLABS says all good. Access via Browser no Problem. Owncloud Client 1.8 says "Das Zertifikat des Ausstellers eines lokal gefundenen Zertifikats konnte nicht gefunden werden
Das oberste Zertifikat der Zertifizierungsstelle ist für diesen Fall nicht vertrauenswürdig". No Problem till last Update. My Cert-Chain: Deutsche Telekom Root CA2 -> DFN -> University

**Comment by dragotin** [comment]
@danimo pls

**Comment by danimo** [comment]
Confirmed on a Wheezy Live CD. Checking for causes now.

**Comment by enricotagliavini** [comment]
Same here when using a self made CA added to the system trusted CA in fedora 21. The CA was added in /etc/pki/ca-trust/source/anchors/ then I run `update-ca-trust extract`  to regenerate the bundle /etc/pki/tls/certs/ca-bundle.crt and co. Verified the certificate is there and checked the validity with

```
enrico@quanthink ~ $ openssl s_client -connect cloud.saurisiamonoi.org:443
CONNECTED(00000003)
depth=1 C = IT, O = Sauri Family, OU = www.saurisiamonoi.org, CN = saurisiamonoi.org CA
verify return:1
depth=0 C = IT, O = Sauri Family, OU = www.saurisiamonoi.org, CN = *.saurisiamonoi.org
verify return:1
---
Certificate chain
 0 s:/C=IT/O=Sauri Family/OU=www.saurisiamonoi.org/CN=*.saurisiamonoi.org
   i:/C=IT/O=Sauri Family/OU=www.saurisiamonoi.org/CN=saurisiamonoi.org CA
---
Server certificate
-----BEGIN CERTIFICATE-----
<cut for brevity>
-----END CERTIFICATE-----
subject=/C=IT/O=Sauri Family/OU=www.saurisiamonoi.org/CN=*.saurisiamonoi.org
issuer=/C=IT/O=Sauri Family/OU=www.saurisiamonoi.org/CN=saurisiamonoi.org CA
---
No client certificate CA names sent
Server Temp Key: ECDH, prime256v1, 256 bits
---
SSL handshake has read 2343 bytes and written 331 bytes
---
New, TLSv1/SSLv3, Cipher is ECDHE-RSA-AES128-GCM-SHA256
Server public key is 4096 bit
Secure Renegotiation IS supported
Compression: NONE
Expansion: NONE
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : ECDHE-RSA-AES128-GCM-SHA256
    Session-ID: 0DBA8117F1136A427F2F5EFD6F10ABE37EDC6F58A894369D13FD1D8A0A836924
    Session-ID-ctx: 
    Master-Key: 9BDEE05C9E5CC19723D19256CA27C1026016597CDC0AD73D9E896669EDA79FDADC261E74254AC7F7784E67302BEC5998
    Key-Arg   : None
    Krb5 Principal: None
    PSK identity: None
    PSK identity hint: None
    TLS session ticket lifetime hint: 300 (seconds)
    TLS session ticket:
    0000 - 49 28 c7 b7 fc 8a fa 80-85 da 03 79 eb 82 a1 62   I(.........y...b
    0010 - e5 9a c5 e3 16 ee b4 9a-f6 bc 59 e3 a0 af 7b 8a   ..........Y...{.
    0020 - 82 bc 44 75 3c b8 2e 3e-80 3e 34 8e 94 bf 58 ea   ..Du<..>.>4...X.
    0030 - a7 63 18 95 97 31 07 af-36 fc 26 16 0a 31 34 a3   .c...1..6.&..14.
    0040 - 9b d8 70 93 d8 87 cc 2b-9f 21 41 0e e0 28 c3 ba   ..p....+.!A..(..
    0050 - 1c 52 e8 4d 41 1b 06 2b-84 e6 30 fd 7e ff 12 0f   .R.MA..+..0.~...
    0060 - 63 40 d7 5b f7 2f 42 12-73 7f 08 59 6f b9 90 aa   c@.[./B.s..Yo...
    0070 - f1 c9 81 58 89 5e 38 42-d1 96 9e e6 8a bb 38 63   ...X.^8B......8c
    0080 - ce 17 92 bc 93 91 6c 98-c8 f0 4d 58 50 1f 07 7b   ......l...MXP..{
    0090 - a1 e6 ad 4a 69 9c a9 3f-d4 59 55 70 40 39 cf 52   ...Ji..?.YUp@9.R
    00a0 - e2 bf bd 89 f5 ae 47 34-7c 59 f7 c5 5c c6 24 cf   ......G4|Y..\.$.

    Start Time: 1426754132
    Timeout   : 300 (sec)
    Verify return code: 0 (ok)
---
```

I'm using the official RPMs provided from the owncloud OBS for fedora 21, not the fedora provided packages. The client log is
```
03-19 09:25:18:301 0x1c71640 "################## ownCloud en_US () 1.8.0"
03-19 09:25:18:301 0x1c71640 void OCC::Application::setupTranslations() Using "en_US" translation
03-19 09:25:18:301 0x1c71640 SocketApi:  server started, listening at  "/run/user/1000/ownCloud/socket"
03-19 09:25:18:302 0x1c71640 ==== added system ignore list to socketapi: "/etc/ownCloud/sync-exclude.lst"
03-19 09:25:18:302 0x1c71640 ==== added user defined ignore list to csync: "/home/enrico/.local/share/data/ownCloud/sync-exclude.lst"
03-19 09:25:18:302 0x1c71640 setting remote poll timer interval to 30000 msec
03-19 09:25:18:328 0x1c71640 * Setup folders from  ""
03-19 09:25:18:328 0x1c71640   ` -> setting up: "Music"
03-19 09:25:18:328 0x1c71640     -> file path:  "/home/enrico/.local/share/data/ownCloud/folders/Music"
03-19 09:25:18:328 0x1c71640 Checked local path ok
03-19 09:25:18:328 0x1c71640 Adding folder to Folder Map  OCC::Folder(0x1f20630)
03-19 09:25:18:329 0x1c71640 (+) Watcher: "/home/enrico/Music/"
03-19 09:25:18:332 0x1c71640     `-> and 39 subdirectories
03-19 09:25:18:332 0x1c71640 Schedule folder  "Music"  to sync!
03-19 09:25:18:332 0x1c71640 Scheduling a sync in 2 seconds
03-19 09:25:18:332 0x1c71640   ` -> setting up: "ownCloud"
03-19 09:25:18:332 0x1c71640     -> file path:  "/home/enrico/.local/share/data/ownCloud/folders/ownCloud"
03-19 09:25:18:332 0x1c71640 Checked local path ok
03-19 09:25:18:332 0x1c71640 Adding folder to Folder Map  OCC::Folder(0x1f20150)
03-19 09:25:18:332 0x1c71640 (+) Watcher: "/home/enrico/ownCloud/"
03-19 09:25:18:333 0x1c71640     `-> and 5 subdirectories
03-19 09:25:18:333 0x1c71640 Schedule folder  "ownCloud"  to sync!
03-19 09:25:18:391 0x1c71640 Sys Info size:  290
03-19 09:25:18:662 0x1c71640 !!! OCC::CheckServerJob created for  QUrl( "https://cloud.saurisiamonoi.org/owncloud" )  querying "status.php"
03-19 09:25:20:332 0x1c71640 FolderMan: Syncing is disabled, no scheduling.
03-19 09:25:22:019 0x1c71640 Client is on latest version!
03-19 09:25:48:302 0x1c71640 void OCC::FolderMan::slotEtagPollTimerTimeout() No folders need to check for the remote ETag
03-19 09:25:50:400 0x1c71640 !!! OCC::CheckServerJob created for  QUrl( "https://cloud.saurisiamonoi.org/owncloud" )  querying "status.php"
03-19 09:25:59:184 0x1c71640 SslHandshakeFailedError:  "SSL handshake failed"  : can be caused by a webserver wanting SSL client certificates
03-19 09:25:59:184 0x1c71640 void OCC::AbstractNetworkJob::slotFinished() 6 "SSL handshake failed"
03-19 09:25:59:184 0x1c71640 AccountState connection status change:  "Undefined" -> "Status not found"
03-19 09:25:59:184 0x1c71640 AccountState state change:  "Disconnected" -> "NetworkError"
03-19 09:25:59:185 0x1c71640 Disabling sync scheduler, terminating sync
03-19 09:25:59:186 0x1c71640 error: status.php replied  0 ""
03-19 09:25:59:237 0x1c71640 "SSL-Errors happened for url  "https://cloud.saurisiamonoi.org/owncloud/status.php"        Error in  QSslCertificate( "3" , "6" , "VerCvFe8b4ri4rKYKdNEKw==" , "Sauri Family" , "Sauri Family" , QMap((1, "saurisiamonoi.org")(1, "*.saurisiamonoi.org")) , QDateTime("Sun Feb 23 16:44:00 2014") , QDateTime("Thu Feb 23 16:44:00 2017") ) : "The issuer certificate of a locally looked up certificate could not be found" ( "The issuer certificate of a locally looked up certificate could not be found" )
        Error in  QSslCertificate( "3" , "6" , "VerCvFe8b4ri4rKYKdNEKw==" , "Sauri Family" , "Sauri Family" , QMap((1, "saurisiamonoi.org")(1, "*.saurisiamonoi.org")) , QDateTime("Sun Feb 23 16:44:00 2014") , QDateTime("Thu Feb 23 16:44:00 2017") ) : "The root CA certificate is not trusted for this purpose" ( "The root CA certificate is not trusted for this purpose" )
        Error in  QSslCertificate( "3" , "6" , "VerCvFe8b4ri4rKYKdNEKw==" , "Sauri Family" , "Sauri Family" , QMap((1, "saurisiamonoi.org")(1, "*.saurisiamonoi.org")) , QDateTime("Sun Feb 23 16:44:00 2014") , QDateTime("Thu Feb 23 16:44:00 2017") ) : "No certificates could be verified" ( "No certificates could be verified" )
 " Certs not trusted by user decision, returning.
03-19 09:25:59:237 0x1c71640 SslHandshakeFailedError:  "SSL handshake failed"  : can be caused by a webserver wanting SSL client certificates
03-19 09:25:59:238 0x1c71640 void OCC::AbstractNetworkJob::slotFinished() 6 "SSL handshake failed"
03-19 09:25:59:238 0x1c71640 error: status.php replied  0 ""
```

**Comment by rfc1459** [comment]
I can confirm the issue on Ubuntu 12.04 as well, but *not* on Ubuntu 14.04.

It appears to be a regression from 1.7.x, given that neither the server configuration nor the client SSL trust store changed.

**Comment by frettloeh** [comment]
I can confirm the same using Ubuntu 12.04 with a thawte DV signed-certificate.

**Comment by danimo** [comment]
This is due to https://bugreports.qt.io/browse/QTBUG-29103 in the upstream version of Qt. I have a workaround, which I will add to the repository as a patch so you can benefit from it in 1.8.0 already.

**Comment by frettloeh** [comment]
1.8.0 is already released, so I suppose, this patch comes into 1.8.1, right? Is there a timeline when this will be released?

**Comment by danimo** [comment]
@frettloeh Nope, I'll add it as a patch to 1.8.0, but it will also be in 1.8.1

**Comment by danimo** [comment]
We just added the patch. It will become available as soon as OBS has buil the packages. If you still see problems after the next update (in ~1-2 hrs), please reopen.

**Comment by frettloeh** [comment]
Thank you for the quick fix which is working under Ubuntu 12.04! :)
